lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1656,6 +1656,17 @@ LogicalResult MultibitMuxOp::canonicalize(MultibitMuxOp op,
     return success();
   }
 
+  // If the index width is narrower than the size of inputs, drop front
+  // elements.
+  auto indexWidth = op.getIndex().getType().getBitWidthOrSentinel();
+  uint64_t inputSize = op.getInputs().size();
+  if (indexWidth < 64 && 1ull << indexWidth < inputSize) {
```

Comment:
Is `1ull` portable across windows and Linux?

Reply:
I think it's fine as there are several existing use cases in LLVM and CIRCT repo.

Reply:
long long is at least 64bits since c++11, according to https://en.cppreference.com/w/cpp/language/types .

That said (only looking at the code added here):

indexWidth is signed (int32_t), and in particular can be negative (!).  Shift by a negative amount is UB! Please add a check for that, as even without UB it's logically not what we want either.